clang/tools/libclang/CMakeLists.txt

```diff
@@ -106,7 +106,8 @@ if (LLVM_EXPORTED_SYMBOL_FILE)
                      DEPENDS ${LIBCLANG_VERSION_SCRIPT_FILE})
 endif()
 
-if(LLVM_ENABLE_PIC OR (WIN32 AND NOT LIBCLANG_BUILD_STATIC))
+if((NOT (WIN32 OR CYGWIN) AND LLVM_ENABLE_PIC) OR
+  ((WIN32 OR CYGWIN) AND NOT LIBCLANG_BUILD_STATIC))
```

Comment:
This bit here is quite unintuitive to read, but I don't have any specific preference for any better way of doing it either...

Reply:
Indeed, that's unfortunate and has always been a problem.
`LLVM_ENABLE_PIC` defaults to ON on Windows, so that condition never worked unless you explicitly disabled `LLVM_ENABLE_PIC` as found out by @jeremyd2019.

Reply:
this stops libclang.dll from building on (non-mingw) windows when we specify `LLVM_ENABLE_PIC`, can we revert that part of this change?

Reply:
Right, so in order not to change the behaviour for other windows platforms, this should probably be `if ((NOT CYGWIN AND LLVM_ENABLE_PIC) OR ((WIN32 OR CYGWIN) AND NOT LIBCLANG_BUILD_STATIC))`, i.e. removing the `WIN32 OR` part from the first part of the conditional?

Reply:
I'm confused now, the condition before had `WIN32 AND NOT LIBCLANG_BUILD_STATIC` but that branch was never hit because `LLVM_ENABLE_PIC` was defaulted `ON` on Windows.  What exactly is `LIBCLANG_BUILD_STATIC` supposed to do, if not switch from building a shared libclang to building a static one?

Reply:
As I recall, the libclang.dll did build successfully so I think that could be reverted.  Perhaps I/we just misunderstood what that option was supposed to do

Reply:
as in,
```
if(LLVM_ENABLE_PIC OR ((WIN32 OR CYGWIN) AND NOT LIBCLANG_BUILD_STATIC))
```

Reply:
sent out #138343